cleaning_transformation.ipynb

```diff
@@ -0,0 +1,70 @@
+import pandas as pd
+import seaborn as sns
+import matplotlib.pyplot as plt
+import csv
+
+import statistics
+
+data = pd.read_csv('Train.csv')
+test_data = pd.read_csv('Test.csv')
+
+#########################################################################
+# reduce data quantity to process with a few
+# this is not used continuously, only to quickly debug code
+sdata = data.iloc[0:5000, :]
+sdata.to_csv('sdata.csv')
+print(sdata)
+#########################################################################
+
+# print columns
+print(data.columns)
+
+# print head set
+print(data.head())
+
+# get shape of the data to visualise observations
+print(data.shape)
+# Dataset consists of 13647309 observations and 48 characteristics
+
+# get description of data
+print(data.describe)
+
+# Examine NaN values across dataframe
+# entire dataframe
+nan_dataframe = data.isna().sum().sum()
+print(f'Total Dataframe nan: {nan_dataframe}')
+
+
+#######  Data Cleansing and Transformation
+
+# strategy to replace nan values
+# the strategy to replace the NaN values is explored with 0 and using mean values
+
+filled_nan_dataframe = data.isna().sum().sum()
+print(f'\nTotal NaN before fillna 0: {filled_nan_dataframe}')
+data = data.fillna(0)

```

Comment:
i think replacing all data with a zero value can lead to data distortion (for example, affect the average salary) and this method does not work with outliers (also like salary and age), but it is good idea to change some columns like 'ult_fec_cli_1t', 'conyuemp' and see how it affect on analysis.